src/wiki.ts

```diff
@@ -306,11 +307,26 @@ async function generateWikiTerraformModule(terraformModule: TerraformModule): Pr
   const changelog = getTerraformModuleFullReleaseChangelog(terraformModule);
   const tfDocs = await generateTerraformDocs(terraformModule);
   const moduleSource = getModuleSource(context.repoUrl, config.useSSHSourceFormat);
+  const latestTag = terraformModule.getLatestTag();
+
+  // Determine the ref value based on config.moduleRefMode
+  let ref = latestTag;
+  if (config.moduleRefMode === MODULE_REF_MODE_SHA) {
+    const commitSHA = terraformModule.getLatestTagCommitSHA();
+    if (commitSHA) {
+      // Use the SHA as the ref value
+      ref = commitSHA;
+    } else {
+      // If we don't have a SHA, fall back to using the tag
+      info(`Warning: No commit SHA found for tag '${latestTag}', using tag as ref`);
+    }
+  }
```

Comment:
can we switch this to a switch and leverage both modes.

Then move the output for the ref is empty as a warning but don't coerce/fallback. just set it to empty string.